test/sql/updates/setup.continuous_aggs.v2.sql

```diff
@@ -263,3 +263,51 @@ BEGIN
 END $$;
 
 REFRESH MATERIALIZED VIEW mat_ignoreinval;
+
+-- test proper invalidations after update
+CREATE TABLE inval_test (time TIMESTAMPTZ, location TEXT, temperature DOUBLE PRECISION);
+SELECT create_hypertable('inval_test', 'time', chunk_time_interval => INTERVAL '1 hour');
+ 
+INSERT INTO inval_test
+SELECT tm.tval, loc.lval, temp 
+FROM (SELECT generate_series(now() - '4 hours'::interval, now(), '30 seconds') AS tval) tm 
+    CROSS JOIN (select 'loc_' || generate_series(1,3,1) AS lval) loc,
+    generate_series(50.0, 70.0, 1443) temp;
+
+DO LANGUAGE PLPGSQL $$
+DECLARE
+  ts_version TEXT;
+BEGIN
+  SELECT extversion INTO ts_version FROM pg_extension WHERE extname = 'timescaledb';
+
+  IF ts_version < '2.0.0' THEN
+    CREATE VIEW mat_inval
+    WITH ( timescaledb.continuous, timescaledb.materialized_only=true,
+           timescaledb.refresh_lag='-2 hour',
+           timescaledb.ignore_invalidation_older_than = '24 hour',
+           timescaledb.refresh_interval='20 minutes' )
+    AS 
+      SELECT time_bucket('10 minute', time) as bucket, location, min(temperature) as min_temp,
+        max(temperature) as max_temp, avg(temperature) as avg_temp
+      FROM inval_test
+      GROUP BY bucket, location;
+  ELSE
+    CREATE MATERIALIZED VIEW mat_inval
+    WITH ( timescaledb.continuous, timescaledb.materialized_only=true )
+    AS 
+      SELECT time_bucket('10 minute', time) as bucket, location, min(temperature) as min_temp, 
+        max(temperature) as max_temp, avg(temperature) as avg_temp
+      FROM inval_test
+      GROUP BY bucket, location WITH NO DATA;
+
+    PERFORM add_continuous_aggregate_policy('mat_inval', '1 day'::interval, '-2 hours'::interval, '20 minutes');
+  END IF;
+END $$;
+
+REFRESH MATERIALIZED VIEW mat_inval;
+
+INSERT INTO inval_test
+SELECT tm.tval, loc.lval, temp 
+FROM (SELECT generate_series(now() + '30 seconds'::interval, now() + '4 hours'::interval, '30 seconds') AS tval) tm 
+    CROSS JOIN (select 'loc_' || generate_series(1,3,1) AS lval) loc,
+    generate_series(70.0, 90.0, 1440) temp;
```

Comment:
ok this query does not look right, the now() will guarantee that this test will be flaky and the `generate_series(70.0, 90.0, 1440)` just returns a single row so you could just replace it by `70.0`

Reply:
See my comment above about now() just being used to generate a set number of rows above the invalidation threshold.  If you still think this will be flaky, I can change this to an arbitrarily far in the future data.  Fixed the generate series.